### ISSUE TITLE
Not execute the .profile file

### ISSUE THREAD

**Issue by futureproperty** [mintty/wsltty#56]
## if I use bash.exe from powershell,it will excute the .profile(I have added "umask 022" line)  
![wsl2](https://user-images.githubusercontent.com/13601454/29726134-3b75a0ee-8a01-11e7-9650-beed06c55d70.PNG)  
## but if use wsltty to launch wsl,it won't excute the .profile,like this:
![wsl1](https://user-images.githubusercontent.com/13601454/29726118-30d0c1aa-8a01-11e7-9e9b-1f69bbe36de8.PNG)

sorry for my bad english

**Comment by mintty** [comment]
Do you use the initial Ubuntu on Windows installation or the newer one, installed from the Windows store?

**Comment by futureproperty** [comment]
I installed wsl from Windows store.@mintty

**Comment by mintty** [comment]
@rprichard, from my tests I have the impression that the wslbridge parameter `-C~` does not work anymore.

**Comment by rprichard** [comment]
I just tested `wslbridge -C~` with new versions of everything (Win10 16257, Windows Store's Ubuntu 16.04.2, and a new Cygwin install), and it worked fine.

FWIW: wslbridge doesn't start a bash login shell by default, but it looks like MS's `bash.exe` and `ubuntu.exe` *do*, so wslbridge should probably change. However, AFAICT wsltty already overrides wslbridge's default behavior by running `bash -l` explicitly, so it shouldn't matter.

I looked at the shortcuts created after installing wsltty 1.7.9, and I saw these command lines:

 * Ubuntu in Mintty: `...\mintty.exe ... /bin/wslbridge --distro-guid "{969399a6-ae69-4323-9733-bf056eb4c1aa}" -t /bin/bash`
 * WSL % in Mintty: `...\mintty.exe ... /bin/wslbridge -t /bin/bash`
 * WSL ~ in Mintty: `...\mintty.exe ... /bin/wslbridge -C~ -t /bin/bash`
 * WSL -l in Mintty: `...\mintty.exe ... /bin/wslbridge -t /bin/bash -l`

Bash will only source the profile/login scripts with wsltty's -l shortcut. wslbridge will change to the WSL home directory for only the ~ shortcut.

Suggestion: maybe all the shortcuts should use `-l`? ("WSL %" and "WSL ~" would then be identical, so maybe keep only one of them.)

**Comment by mintty** [comment]
My confusion, sorry. -C~ is not supposed to invoke a login shell, nor are two of the three WSL shortcuts.
@futureproperty, which shortcut did you use?

This was designed on purpose to match the default behaviour of the original "Bash on Windows" WSL which did not invoke a login shell either. Now it seems that for the Ubuntu store WSL, the Windows bash.exe launcher seems to invoke a login shell. Hmm, not sure, what that means for the shortcut parameters that should be used. Is it dependent on the distribution; could it again be different for a different one; is it true at all? (to be checked)

**Comment by futureproperty** [comment]
@mintty ,I used the shortcut in Start Menu. 
"Start Menu\Programs\WSLtty\WSL ~ in Mintty.lnk"

**Comment by mintty** [comment]
Obviously the bash.exe Windows launcher changed behaviour from the Legacy version (where it did not start a login shell as far as I remember), now it starts a login shell (as @rprichard already noted), so wsltty should change that.
This will also reduce the number of shortcuts/scripts to be installed and the associated potential confusion.
I can do that with a `-l` parameter, but maybe a login option in wslbridge would also be good, so $0 can be set to `-bash` as some people may expect to see.

**Comment by rprichard** [comment]
@mintty At the least, wslbridge should default to invoking bash as `-bash`. Do you think it should simply have an option to specify argv0 (e.g. https://github.com/rprichard/wslbridge/issues/9)?

**Comment by mintty** [comment]
I see no need to set an arbitrary $0. There should rather be a login option that applies the traditional Unix way, by prepending '-' to the invoked program name (basename) as $0. And I would not make it the default, to stay flexible and be consistent with other programs.

**Comment by mintty** [comment]
Released 1.8.0 which does not request a particular shell anymore, so it does not manipulate the startup sequence.

**Comment by ksze** [comment]
I don't understand the solution to this ticket.
I installed wsltty 1.8.1. By default, I still don't get a login shell that sources .profile when I launch the "Ubuntu Terminal" shortcut (the one without the "%" in the shortcut name). Am I supposed to modify the shortcut?

The "Target" of the shortcut currently looks like this:
```
%LOCALAPPDATA%\wsltty\bin\mintty.exe --WSL="Ubuntu" -h err --configdir="%APPDATA%\wsltty" -~ 
```

What am I supposed to change it to, so that I get a login shell?

**Comment by mintty** [comment]
You could add an explicit invocation `/bin/wslbridge -t /bin/bash -l` or wait for wsltty 1.8.2 (planned soon).

**Comment by ksze** [comment]
@mintty Where would I add the invocation? I tried to append `/bin/wslbridge -t /bin/bash -l` to the arguments of the "WSL Terminal" shortcut, but that just causes mintty to close immediately.

**Comment by mintty** [comment]
Hmm. Apparently while optimizing the launch to need fewer parameters and support login mode (if available), the option to use the full invocation form was lost, sorry.

**Comment by mintty** [comment]
Actually, it now works without referring to /bin/wslbridge explicitly:
`...\mintty.exe --WSL=... --configdir=... -~ /bin/bash -l`

**Comment by mintty** [comment]
Released 3.1.0.
